sensors/ObstacleSensor.py

```diff
@@ -0,0 +1,53 @@
+from sensors.SensorCommon import Sensor
+from sensors.SensorCommon import Dist2Circ
+import numpy as np
+import operator
+
+
+class ObstacleSensor(Sensor):
+
+    def __init__(self, obstacles=False, nbObstacles=0, SensorRange=10, mode="position"):
+        super().__init__(nbObservations=nbObstacles, SensorRange=SensorRange)
+        ## todo: implement obstacle and goal flag.
+        self._Observation = np.ones([self._nbObservations, 2]) * self._limSensor
+        self._mode = mode
+        self._setSensorName()
+
+    def _setSensorName(self):
+        if self._mode == "position":
+            self._name = "ObstaclePseudoSensorPosition"
+        elif self._mode == "distance":
+            self._name = "ObstaclePseudoSensorDistance"
+
+    def name(self):
+        return self._name
+
+    def _reset(self):
+        self._Observation[:] = self._limSensor
+
+    def sense(self, s, goals, obstacles, t=0):
+        self._reset()
+        if self._mode == "position":
+            for idx, obst in enumerate(obstacles):
+                if idx >= self._nbObservations:
+                    break
+                self._Observation[idx][0] = obst.position(t=t)[0]
+                self._Observation[idx][1] = obst.position(t=t)[1]
```

Comment:
Do you need to explicitly split between x and y coordinate?

Reply:
no not at all. I can pass them directly.